src/include/txn.i

```diff
@@ -193,6 +193,166 @@ __wt_timestamp_subone(wt_timestamp_t *ts)
 
 #endif /* HAVE_TIMESTAMPS */
 
+/*
+ * __txn_op_modify_recno --
+ *      Modify the latest txn op with the given recno.
+ */
+static inline void
+__txn_op_modify_recno(WT_SESSION_IMPL *session, uint64_t recno)
+{
+	WT_TXN *txn;
+	WT_TXN_OP *op;
+
+	txn = &session->txn;
+
+	WT_ASSERT(session, txn->mod_count > 0 && recno != WT_RECNO_OOB);
+	op = txn->mod + txn->mod_count - 1;
+
+	WT_ASSERT(session, op->type == WT_TXN_OP_BASIC_COL ||
+	    op->type == WT_TXN_OP_INMEM_COL);
+
+	op->u.op_col.recno = recno;
+}
+
+/*
+ * __txn_prepared_op_resolve --
+ *      Resolve a transaction operation indirect references.
+ * In case of prepared transactions, the prepared updates could be evicted using
+ * cache overflow mechanism. Transaction operations referring these prepared
+ * updates would be referring to them using indirect references
+ * (i.e keys/recnos), which need to be resolved as part of that transaction
+ * commit/rollback.
+ */
+static inline int
+__txn_prepared_op_resolve(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit)
+{
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_TXN *txn;
+	WT_UPDATE *upd;
+	const char *open_cursor_cfg[] = {
+	    WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL };
+
+	txn = &session->txn;
+
+	switch (op->type) {
```

Comment:
This switch doesn't look worthwhile to me. Could you do:
```
if (op->type == WT_TXN_OP_NONE || op->type == WT_TXN_OP_REF_DELETE || op->type == WT_TXN_OP_TRUNCATE_COL || op->type == WT_TXN_OP_TRUNCATE_ROW)
    return (0);
```
Then remove the switch? That allows you to remove a level of indenting as well.